packages/popover/src/Popover.types.ts

```diff
@@ -11,6 +11,39 @@ type TransitionLifecycleCallbacks = Pick<
   'onEnter' | 'onEntering' | 'onEntered' | 'onExit' | 'onExiting' | 'onExited'
 >;
 
+/**
+ * Options to render the popover element
+ * @param Inline will render the popover element inline with the reference element
+ * @param Portal will render the popover element in a provided `portalContainer` or
+ * in a new div appended to the body
+ * @param TopLayer will render the popover element in the top layer
+ */
+export const RenderMode = {
+  Inline: 'inline',
+  Portal: 'portal',
+  TopLayer: 'top-layer',
+} as const;
+export type RenderMode = (typeof RenderMode)[keyof typeof RenderMode];
+
+/**
+ * Options to control how the popover element is dismissed. This should not be extended
+ * because it is intended to have parity with the web-native {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/popover popover attribute}
+ * @param Auto will automatically handle dismissal on backdrop click or esc key press, ensuring only one popover is visible at a time
+ * @param Manual will require that the consumer handle dismissal manually
+ */
+export const DismissMode = {
+  Auto: 'auto',
+  Manual: 'manual',
+} as const;
+export type DismissMode = (typeof DismissMode)[keyof typeof DismissMode];
+
+/** Local implementation of web-native `ToggleEvent` until we use typescript v5 */
+export interface ToggleEvent extends Event {
```

Comment:
I don't think this extends `Event`, does it? (i.e. it doesn't have `e.target` etc. iirc)